src/Data/Vec/Binary/Base.agda

```diff
@@ -0,0 +1,104 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- Vectors with faster indexing based on binary numbers
+------------------------------------------------------------------------
+
+{-# OPTIONS --cubical-compatible --safe #-}
+
+module Data.Vec.Binary.Base where
+
+open import Data.Fin.Binary.Base as Finᵇ hiding (zero; suc)
+open import Data.Nat.Binary.Base
+open import Function.Base using (_∘_)
+open import Level using (Level)
+
+private
+  variable
+    a b c : Level
+    A : Set a
+    B : Set b
+    C : Set c
+    n : ℕᵇ
+
+-- Core type
+------------------------------------------------------------------------
+
+data Vecᵇ (A : Set a) : ℕᵇ → Set a where
+  -- The empty vector
+  []        :                               Vecᵇ A zero
+  -- A vector with an odd number of elements
+  _∷⟨_/_⟩   : A     → Vecᵇ A n → Vecᵇ A n → Vecᵇ A 1+[2 n ]
+  -- A vector with a non-zero even number of elements
+  _×_∷⟨_/_⟩ : A → A → Vecᵇ A n → Vecᵇ A n → Vecᵇ A 2[1+ n ]
```

Comment:
The `×` in `_×_∷⟨_/_⟩` looks a bit weird to me, because `×` is usually used at the type level. I would argue for `_∷_∷⟨_/_⟩` instead?